janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/BufferUtil.java

```diff
@@ -57,6 +65,50 @@ public static StaticBuffer getIntBuffer(int[] ids) {
         return StaticArrayBuffer.of(arr);
     }
 
+    /**
+     * We always add a byte marker at the beginning to indicate it's a string buffer. Then we
+     * write the string uncompressed. Finally, if the total buffer size is 8, we add a dummy byte
+     * at the end so that upon read time, JanusGraph knows the buffer does not store a long value.
+     * See {@link IDManager#getKeyID(StaticBuffer)} for more details.
+     *
+     * Note that we apply STOP_MASK to the last character. We don't have to do it because a static buffer
+     * has a fixed length, and thus upon read time, we know where to stop reading the string. This is more
+     * to keep it consistent with {@link StringEncoding#writeAsciiString(byte[], int, String)} where we use
+     * STOP_MASK to mark the end of the string. An additional benefit of doing so is to enable corruption check
+     * in the future. Note: this approach has no overhead.
+     *
+     * @param s
+     * @return
+     */
+    public static StaticBuffer getStringBuffer(String s) {
+        if (StringUtils.isEmpty(s) || !StringUtils.isAsciiPrintable(s)) {
+            throw new IllegalArgumentException("value must be non-empty printable ASCII string but received: " + s);
+        }
+        ByteBuffer buffer;
```

Comment:
(nitpick)
Doesn't look like any of `ByteBuffer`  specific functionality is used in this method. I.e. you could potentially used simple `byte[]` instead. That said, I'm fine with `ByteBuffer`.

Reply:
I noticed our code base has been using `ByteBuffer` consistently even if as you said, many places could have just used plain byte array.

I am guessing the reason is that if we decide to switch to off-heap memory, we can do so without too many code changes.

Reply:
That’s a good point. I think we should use ByteBuffer as you already do in such case just for consistency reasons.